smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -83,7 +88,11 @@ contract MixinPurchase is
       toKey.tokenId
     );
 
-    // We explicitly allow for greater amounts of ETH to allow 'donations'
+    // We explicitly allow for greater amounts of ETH or tokens to allow 'donations'
+    if(_value > 0) {
```

Comment:
So from your last comments, `_value` will always be greater than 0 (with the exception of purchasing a free key) if we force the user to always specify the exact `_value`, right?

Reply:
Yea - give me a bit to make those updates.  Fixing the tests will be a bit tedious.